compaction/compaction.hh

```diff
@@ -100,12 +101,28 @@ struct compaction_result {
     compaction_stats stats;
 };
 
+class read_monitor_generator;
+
+class compaction_read_monitor_generator_wrapper {
+    std::unique_ptr<read_monitor_generator> _generator = nullptr;
+    uint64_t _progress = 0;
+public:
+    void set_generator(std::unique_ptr<read_monitor_generator> generator);
+    void reset_generator();
+    uint64_t get_progress() const;
```

Comment:
Please document get_progress(), as it's not clear what's the unit of progress here.